ibis/backends/pandas/client.py

```diff
@@ -157,7 +157,7 @@ def _infer_pandas_series_contents(s: pd.Series) -> dt.DataType:
     """
     if s.dtype == np.object_:
         inferred_dtype = infer_pandas_dtype(s, skipna=True)
-        if inferred_dtype in {'mixed', 'decimal'}:
```

Comment:
`'decimal'` being a special case here was a carry-over from old code. Should be OK to remove the special case, which would make it just use the type mapping above (L82, `_inferable_pandas_dtypes`)